gr-blocks/lib/count_bits.cc

```diff
@@ -45,16 +41,18 @@ unsigned int count_bits16(unsigned int x)
 
 unsigned int count_bits32(unsigned int x)
 {
-    unsigned res = (x & 0x55555555) + ((x >> 1) & 0x55555555);
-    res = (res & 0x33333333) + ((res >> 2) & 0x33333333);
-    res = (res & 0x0F0F0F0F) + ((res >> 4) & 0x0F0F0F0F);
-    res = (res & 0x00FF00FF) + ((res >> 8) & 0x00FF00FF);
-    return (res & 0x0000FFFF) + ((res >> 16) & 0x0000FFFF);
+    unsigned res = 0;
+    volk_32u_popcnt(&res, x);
+
+    return res;
 }
 
 unsigned int count_bits64(unsigned long long x)
 {
-    return count_bits32((x >> 32) & 0xffffffff) + count_bits32(x & 0xffffffff);
+    uint64_t res_as_u64 = 0;
+    volk_64u_popcnt(&res_as_u64, static_cast<uint64_t>(x));
+
+    return static_cast<unsigned long long>(res_as_u64);
```

Comment:
hm, but you're return an `unsigned int`, not an `unsigned long long`; and frankly, that should suffice for the near future :)

Reply:
While we're touching this, can you please change this to uint64_t?

Reply:
Sorry I meant the function argument, not the return value.

Reply:
What, where did that odd mistake come from...I'll make these changes when I get home from work.

Reply:
I'd argue the return type is also wrong. Should `count_bits64` be returning a 32-bit number?

Reply:
Derp, ignore my uint64_t comment, I was thinking byteswap for some reason.